### PR TITLE
Delete TcpConnectionFactory::createDuplexConnectionFromSocket()

### PR DESCRIPTION
**Pull request by alexmalyshev**
It's a very verbose way of doing std::make_unique\<TcpDuplexConnection\>(...)

### CODE REVIEW

File: src/transports/tcp/TcpConnectionFactory.cpp [hunk 1]
```diff
@@ -80,12 +80,4 @@ void TcpConnectionFactory::connect(OnDuplexConnectionConnect cb) {
       });
 }
 
-std::unique_ptr<DuplexConnection>
-TcpConnectionFactory::createDuplexConnectionFromSocket(
```
Comment (phoad) [inline src/transports/tcp/TcpConnectionFactory.cpp]: I have deleted this function in my previous PR and @lehecka has informed that this function needs to stay. I am not sure if the concern of him still exists.

**Comment by lehecka** [comment]
TcpDuplexConnection is supposed to be an internal type, not part of the public API. The idea is that TcpConnectionFactory::createDuplexConnectionFromSocket is returning DuplexConnection base type and hiding the way how is the instance created.

**Comment by alexmalyshev** [comment]
> TcpDuplexConnection is supposed to be an internal type, not part of the public API.

I'd argue that if it's being returned at all to the user's code and they have to use it, then it's part of the public API.  No point with bothering with passthrough functions.

**Comment by lehecka** [comment]
To be clear what I meant:
TcpDuplexConnection is not returned to the application code. It is returned as DuplexConnection (the base type).
There is nothing interesting TcpDuplexConnection so I would rather keep it "invisible" and not allow application code to depend on this type.

**Comment by alexmalyshev** [comment]
Fair enough.  I didn't realize it was actually being returned as a `unique_ptr<DuplexConnection>`.

**Pull request closed without merging**